### ISSUE TITLE
--base-url option should be consistent across subcommands

### ISSUE THREAD

**Issue by rootkea** [getzola/zola#1466]
Hello!

In general, options should be consistent across the subcommands. But -u is not.

```
$ zola build -u 
error: The argument '--base-url <base_url>' requires a value but none was supplied

USAGE:
    zola build --base-url <base_url>

For more information try --help
$ zola serve -u 
Building site...
-> Creating 2 pages (0 orphan), 1 sections, and processing 0 images
Done in 25ms.

Web server is available at http://127.0.0.1:1111

Listening for changes in blog{config.toml, content, sass, static, templates}
Press Ctrl+C to stop

^C$
```

**Comment by Keats** [comment]
Yep, we should remove the base_url argument for serve, it's not used anyway.

**Comment by rootkea** [comment]
Umm if we supply argument to `--base-url`  then the arg provided seems to be used by `page.permalink | safe`.

```
$ zola serve -u foo.bar
```
Same happens for `zola build -u foo.bar`.

The issue is when `--base-url` option lacks the argument. For that case `zola build -u` given an error while `zola serve -u` succeeds which is inconsistent behavior.

**Comment by Keats** [comment]
Ah I think for `zola serve` it's there for some docker based environments or something.

**Comment by rootkea** [comment]
Fixed by #1474
